src/permissions.c

```diff
@@ -26,6 +26,55 @@
 
 static XdpDbusImplPermissionStore *permission_store = NULL;
 
+static gboolean
+maybe_check_base_snap_permissions (const char   *app_id,
+                                   const char   *table,
+                                   const char   *id,
+                                   char       ***out_permissions)
+{
+  const char *app_name_part;
+
+  /* We used to check permissions for the base snap, so if those were granted
+   * already we should just return those, migrating to the new name.
+   * But we do it only for the main application.
+   */
+  if (g_str_has_prefix (app_id, "snap.") &&
```

Comment:
Some functions here would make it much clearer what all the string processing is doing.